dhis2/dhis2-core
# fix: cors config in spring 6

### Issue
- In Spring 5 we enable `corsFilter` by adding a filter as below. Its not working in Spring security 6.
`http.addFilterBefore(DhisCorsFilter.get(), BasicAuthenticationFilter.class);`


### Fix
- Use `http.cors(Customizer.withDefaults());` to enable cors. 
- Spring Security will look for a bean name `corsFilter` which we also defined in security configs with our custom `DhisCorsProcessor`


### Test
- Still not figure out how to write test for this yet.
- Can manually test with below steps:
1) Use an online tool at `https://cors-test.codehappy.dev`
2) Can test now by click on `Test` button, it will receive error from Dhis2 server because we haven't added the website to our cors allow list.
3) Log in Dhis2 instance > SystemSettings > Access : add the above url to the cors allow list.
4) Back to the cors test website, click on button `Test`
5) Verify the return response, it should contain: `access-control-allow-origin: https://cors-test.codehappy.dev/`
